specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/kustoPool.json

```diff
@@ -544,49 +607,44 @@
               "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
             }
           }
-        },
-        "x-ms-long-running-operation": true
-      },
-      "delete": {
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/kustoPools/{kustoPoolName}/listLanguageExtensions": {
```

Comment:
if an entity has a "list", "add", and "remove" post action it should be modeled as a GET/PUT/DELETE proxy resource.  Is there a reason not to do so here?

Reply:
Those are the action we introduce in Kusto swagger. 
The reason we choose this solution is because language extension cannot be configured in the creation of the cluster/KustoPool. The cluster need to be created and only then you can add language extension. That's why we didn't add it as a property to the cluster/KustoPool resource. 

Since it is already deployed in our backend, I prefer that we stick to those actions and we can discuss deprecating them and moving to proxy sub-resource (or another solution) offline. What do you think?

Reply:
Yes, these should be child resources instead of POST actions so users can completely configure their cluster/pool via a template deployment.  Please fix this in a future version